### PR TITLE
DDL hardening

### PR DESCRIPTION
**Pull request by nirbhayc**
- Comment unused parser constructs
- Force PKs columns NOT NULL
- Explicit check for multiple PK definitions

### CODE REVIEW

**Comment by nirbhayc** [comment]
All failing tests passed on my system except for qcopy (which tries to exec copycomdb2, not in PATH).

File: sqlite/comdb2build.c [hunk 1]
```diff
@@ -2311,6 +2312,13 @@ static char *gen_key_name(struct comdb2_ddl_context *ctx, const char *tabname,
     return keyname;
 }
 
+#define PK "PRIMARY_KEY"
```
Comment (akshatsikarwar) [inline sqlite/comdb2build.c]: Consider using `COMDB2_PK` or similar to avoid name clashes with existing instances.

Comment (nirbhayc) [inline sqlite/comdb2build.c]: Sure

Comment (nirbhayc) [inline sqlite/comdb2build.c]: @akshatsikarwar thought you were suggesting to rename the proprocessor identifier "PK". How about prefixing with a $, "$PRIMARY_KEY"?

**Pull request merged**